cmd/bosun/sched/check.go

```diff
@@ -185,6 +186,44 @@ func (s *Schedule) RunHistory(r *RunHistory) {
 	if checkNotify && s.nc != nil {
 		s.nc <- true
 	}
+	// Instrument alert states.
+	severityCount := make(map[string]int64)
+	lastAbnormalCount := make(map[string]int64)
+	unAckedCount := map[bool]int64{true: 0, false: 0}
+	activeCount := map[bool]int64{true: 0, false: 0}
+	for _, state := range s.status {
+		if !state.Open {
+			return
+		}
+		severity := state.Status().String()
+		if _, ok := severityCount[severity]; ok {
+			severityCount[severity]++
+		} else {
+			severityCount[severity] = 1
+		}
+		lastAbnormal := state.AbnormalStatus().String()
+		if _, ok := lastAbnormalCount[lastAbnormal]; ok {
+			lastAbnormalCount[lastAbnormal]++
+		} else {
+			lastAbnormalCount[lastAbnormal] = 1
+		}
+		unAckedCount[state.NeedAck]++
+		activeCount[state.IsActive()]++
+	}
+	for k, v := range severityCount {
+		collect.Put("alerts.current_severity", opentsdb.TagSet{"severity": k}, v)
+	}
+	metadata.AddMetricMeta("bosun.alerts.current_severity", metadata.Gauge, metadata.Alert, "The number of open alerts by current severity.")
+	for k, v := range lastAbnormalCount {
+		collect.Put("alerts.last_abnormal_severity", opentsdb.TagSet{"severity": k}, v)
+	}
+	metadata.AddMetricMeta("bosun.alerts.last_abnormal_severity", metadata.Gauge, metadata.Alert, "The number of open alerts by last abnormal severity.")
```

Comment:
Would it be a horrible idea for some of these to include alert name in the tags and emit one point per alert? Could be cool to graph status of an individual alert over history.